src/lib/broadcast.jl

```diff
@@ -55,6 +56,7 @@ unbroadcast(x::AbstractArray, x̄) =
 unbroadcast(x::Number, x̄) = accum_sum(x̄)
 unbroadcast(x::Tuple{<:Any}, x̄) = (accum_sum(x̄),)
 unbroadcast(x::Base.RefValue, x̄) = (x=accum_sum(x̄),)
+unbroadcast(x::Tuple, x̄) = trim(x, length(x) == length(x̄) ? x̄ : accum_sum(x̄; dims=2:ndims(x̄))) # case length(x) > 1
```

Comment:
I wonder if we can use dispatch on abstractarrays instead of this branching

Reply:
What this should avoid is summing a 9x1 matrix over the 2nd dimension, and hence making a copy. whether this can plausibly matter for arrays short enough to become tuples, I don't know. The function does the same thing for arrays (just above) where it can matter:
```
julia> @btime sum($(rand(100,1,100)); dims=2);
  3.713 μs (8 allocations: 78.31 KiB)

julia> @btime sum($(rand(100,2,100)); dims=2);
  5.229 μs (8 allocations: 78.31 KiB)

julia> @btime reshape($(rand(100,1,100)), (100,100));
  17.117 ns (1 allocation: 64 bytes)
```